### PR TITLE
Family search should return at most the requested number of unique results

### PR DESCRIPTION
**Pull request by marta-**
Plus some variable naming and documentation fixes.

### CODE REVIEW

**Comment by marta-** [comment]
@allasm please review this first thing on Monday. Thanks!

File: components/family-studies/api/src/main/java/org/phenotips/studies/family/internal/export/FamilySearchResult.java [hunk 1]
```diff
@@ -71,6 +73,21 @@ public FamilySearchResult(Patient patient, boolean usePatientName, Family family
 
     }
 
+    @Override
+    public boolean equals(Object other)
```
Comment (veronikaslc) [inline components/family-studies/api/src/main/java/org/phenotips/studies/family/internal/export/FamilySearchResult.java]: I can't find where exactly we use it. If we don't need it, why we have it?

**Comment by veronikaslc** [comment]
I don't see how/where exactly we avoid duplications in this PR or in the code in general.

File: components/family-studies/api/src/main/java/org/phenotips/studies/family/internal/export/PhenotipsFamilyExport.java [hunk 2]
```diff
@@ -213,12 +214,12 @@ private void queryFamilies(String input, String requiredPermissions, int results
...
+            results.add(new FamilySearchResult(family, requiredPermissions));
         }
     }
 
     private void queryPatients(String input, String requiredPermissions, int resultsLimit,
-        List<FamilySearchResult> resultsList)
+        Set<FamilySearchResult> results)
     {
         StringBuilder querySb = new StringBuilder();
         querySb.append("from doc.object(PhenoTips.PatientClass) as patient, ");
```
Comment (veronikaslc) [inline components/family-studies/api/src/main/java/org/phenotips/studies/family/internal/export/PhenotipsFamilyExport.java]: More of a side question, but why do we need next line?

File: components/family-studies/api/src/main/java/org/phenotips/studies/family/internal/export/FamilySearchResult.java (hunk 1 above)
Comment (marta-) [inline components/family-studies/api/src/main/java/org/phenotips/studies/family/internal/export/FamilySearchResult.java]: You always need to override equals & hashCode for any class that is used in a Set or as key in a Map.

**Comment by marta-** [comment]
@veronikaslc :
> I don't see how/where exactly we avoid duplications in this PR or in the code in general.
By using a Set instead of a List.

File: components/family-studies/api/src/main/java/org/phenotips/studies/family/internal/export/PhenotipsFamilyExport.java (hunk 2 above)
Comment (marta-) [inline components/family-studies/api/src/main/java/org/phenotips/studies/family/internal/export/PhenotipsFamilyExport.java]: Right, might be redundant. Doesn't break anything ATM tho.

Comment (veronikaslc) [inline components/family-studies/api/src/main/java/org/phenotips/studies/family/internal/export/PhenotipsFamilyExport.java]: We can potentially just search for patients whose families are not included in the results from family search.

Comment (marta-) [inline components/family-studies/api/src/main/java/org/phenotips/studies/family/internal/export/PhenotipsFamilyExport.java]: We could, there are many ways of implementing this. This one seemed to me the one easiest to maintain.

**Pull request merged**